CMakeLists.txt

```diff
@@ -592,223 +606,249 @@ install(TARGETS helics_base EXPORT helics-targets)
-      "HELICS Github"
-      "https://gmlc-tdc.github.io/HELICS-src/"
-      "Helics Documentation"
-      "https://www.helics.org"
-      "Helics Web page"
-      "https://www.youtube.com/channel/UCPa81c4BVXEYXt2EShTzbcg"
-      "TDC YouTube channel"
+        CPACK_COMPONENTS_ALL
+        applications
+        headers
+        libs
+        runtime
+        matlab
+        python
+        java
+        octave
+        csharp
     )
-  else(WIN32)
-	set(CPACK_PACKAGE_ICON "${CMAKE_SOURCE_DIR}/docs/img/HELICS.ico")
-  endif(WIN32)
-
-  if(APPLE)
-    set(CPACK_BUNDLE_NAME "libhelics")
-    configure_file(
-      "${PROJECT_SOURCE_DIR}/config/Info.plist.in"
-      "${CMAKE_CURRENT_BINARY_DIR}/Info.plist" @ONLY
+    if(WIN32)
+        set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}\\\\LICENSE")
+    else(WIN32)
+        set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}/LICENSE")
+    endif(WIN32)
+
+    set(CPACK_COMPONENT_APPLICATIONS_DISPLAY_NAME "Application")
+    set(CPACK_COMPONENT_LIBS_DISPLAY_NAME "Libraries")
+    set(CPACK_COMPONENT_HEADERS_DISPLAY_NAME "Headers")
+    set(CPACK_COMPONENT_RUNTIME_DISPLAY_NAME "Runtime Libraries")
+
+    set(CPACK_COMPONENT_MATLAB_GROUP interfaces)
+    set(CPACK_COMPONENT_JAVA_GROUP interfaces)
+    set(CPACK_COMPONENT_OCTAVE_GROUP interfaces)
+    set(CPACK_COMPONENT_PYTHON_GROUP interfaces)
+    set(CPACK_COMPONENT_CSHARP_GROUP interfaces)
+
+    set(
+        CPACK_COMPONENT_APPLICATION_DESCRIPTION
+        "Executables and helper applications for HELICS"
+    )
+    set(
+        CPACK_COMPONENT_LIBS_DESCRIPTION
+        "Libraries for compiling and linking with HELICS"
     )
-    set(CPACK_BUNDLE_PLIST "${CMAKE_CURRENT_BINARY_DIR}/Info.plist")
-    set(CPACK_BUNDLE_ICON "${PROJECT_SOURCE_DIR}/docs/img/HELICS.ico")
-  endif(APPLE)
+    set(
+        CPACK_COMPONENT_HEADERS_DESCRIPTION
+        "Headers for linking and compiling with HELICS"
+    )
+    set(CPACK_COMPONENT_RUNTIME_DESCRIPTION "Runtime libraries for HELICS")
 
-  set(CPACK_SOURCE_IGNORE_FILES "/Build*/;/build*/;/.git/")
-  # THIS LINE MUST BE LAST
-  include(CPack)
+    set(
+        CPACK_COMPONENT_GROUP_INTERFACES_DESCRIPTION
+        "additional language interfaces for HELICS"
+    )
+
+    set(CPACK_COMPONENT_LIBS_DEPENDS headers)
+    set(CPACK_COMPONENT_RUNTIME_REQUIRED ON)
+
+    set(
+        CPACK_PACKAGE_EXECUTABLES
+        "helics_broker"
+        "Helics Broker"
+        "helics_app"
+        "Helics app executable"
+        "helics_recorder"
+        "Helics recorder"
+        "helics_player"
+        "Helics Player"
+    )
+
+    if(INSTALL_SYSTEM_LIBRARIES)
+        include(InstallRequiredSystemLibraries)
+    endif()
+
+    if(WIN32)
+        set(
+            CPACK_PACKAGE_ICON
+            "${CMAKE_SOURCE_DIR}\\\\docs\\\\img\\\\HELICS.ico"
+        )
+        set(CPACK_NSIS_MUI_ICON "${CMAKE_SOURCE_DIR}/docs/img/HELICS.ico")
+        set(CPACK_NSIS_INSTALL_ROOT "C:\\\\local\\\\")
```

Comment:
This should be `set(CPACK_NSIS_INSTALL_ROOT "C:\\\\local")` -- there is already an ending slash added by default